projects/plugins/boost/app/assets/src/js/features/notice/context.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useContext, useState, useCallback, ReactNode } from 'react';
+
+type Notice = {
+	id: string;
+	type: 'success' | 'error' | 'pending';
+	message: string;
+};
+
+type NoticeContextType = {
+	notices: Notice[];
+	addOrUpdateNotice: ( notice: Notice ) => void;
+	removeNotice: ( id: string ) => void;
+};
+
+const NoticeContext = createContext< NoticeContextType | undefined >( undefined );
+
+type NoticeProviderProps = {
+	children: ReactNode;
+};
+
+export const NoticeProvider = ( { children }: NoticeProviderProps ) => {
+	const [ notices, setNotices ] = useState< Notice[] >( [] );
```

Comment:
If notices are identified by a string key, wouldn't storing them in an object by key be far simpler?

Reply:
Thanks for the suggestion! Added in e34717d.